--- volume/drivers/netapp/dataontap/client/client_base.py
+++ volume/drivers/netapp/dataontap/client/client_base.py
@@ -42,6 +42,8 @@
         certificate_file = kwargs['certificate_file']
         ca_certificate_file = kwargs['ca_certificate_file']
         certificate_host_validation = kwargs['certificate_host_validation']
+        ssl_cert_path = kwargs.get('ssl_cert_path')
+        ssl_cert_verify = kwargs.get('ssl_cert_verify', True)
         if private_key_file and certificate_file and ca_certificate_file:
             self.connection = netapp_api.NaServer(
                 host=host,
@@ -51,7 +53,8 @@
                 certificate_file=certificate_file,
                 ca_certificate_file=ca_certificate_file,
                 certificate_host_validation=certificate_host_validation,
-                api_trace_pattern=api_trace_pattern)
+                api_trace_pattern=api_trace_pattern,
+                ssl_cert_verify=ssl_cert_verify)
         elif private_key_file and certificate_file:
             self.connection = netapp_api.NaServer(
                 host=host,
@@ -60,7 +63,8 @@
                 private_key_file=private_key_file,
                 certificate_file=certificate_file,
                 certificate_host_validation=certificate_host_validation,
-                api_trace_pattern=api_trace_pattern)
+                api_trace_pattern=api_trace_pattern,
+                ssl_cert_verify=ssl_cert_verify)
         else:
             self.connection = netapp_api.NaServer(
                 host=host,
@@ -68,7 +72,9 @@
                 port=kwargs['port'],
                 username=username,
                 password=password,
-                api_trace_pattern=api_trace_pattern)
+                api_trace_pattern=api_trace_pattern,
+                ssl_cert_path=ssl_cert_path,
+                ssl_cert_verify=ssl_cert_verify)
 
         self.ssh_client = self._init_ssh_client(host, username, password)
 
