--- compute/api.py
+++ compute/api.py
@@ -761,7 +761,8 @@ class API:
 
         # Target disk is a volume. Don't check flavor disk size because it
         # doesn't make sense, and check min_disk against the volume size.
-        if root_bdm is not None and root_bdm.is_volume:
+        # Also, skip checking instance snapshots without base images.
+        if root_bdm is not None and root_bdm.is_volume and root_bdm.source_type != 'snapshot':
             # There are 2 possibilities here:
             #
             # 1. The target volume already exists but bdm.volume_size is not
@@ -4162,7 +4163,7 @@ class API:
     # finally split resize and cold migration into separate code paths
     @block_extended_resource_request
     @block_port_accelerators()
-    @block_accelerators()
+    @block_accelerators(until_service=SUPPORT_ACCELERATOR_SERVICE_FOR_REBUILD)
     @check_instance_lock
     @check_instance_state(vm_state=[vm_states.ACTIVE, vm_states.STOPPED])
     @check_instance_host(check_is_up=True)
