--- policies/__init__.py
+++ policies/__init__.py
@@ -40,6 +40,7 @@ from nova.policies import instance_usage
 from nova.policies import ips
 from nova.policies import keypairs
 from nova.policies import limits
+from nova.policies import live_resize
 from nova.policies import lock_server
 from nova.policies import migrate_server
 from nova.policies import migrations
@@ -98,6 +99,7 @@ def list_rules():
         ips.list_rules(),
         keypairs.list_rules(),
         limits.list_rules(),
+        live_resize.list_rules(),
         lock_server.list_rules(),
         migrate_server.list_rules(),
         migrations.list_rules(),
