--- skyline_apiserver/types/constants.py
+++ skyline_apiserver/types/constants.py
@@ -24,7 +24,7 @@
 
 # request_id middleware will set this into openstack.global_request_id environ
 INBOUND_HEADER = "X-Openstack-Request-Id"
-INBOUND_HEADER_REGEX = "^req-\\w{8}(-\\w{4}){3}-\\w{12}"
+INBOUND_HEADER_REGEX = "^$|^req-\\w{8}(-\\w{4}){3}-\\w{12}"
 
 ERR_MSG_TOKEN_REVOKED = "The token has revoked."
 ERR_MSG_TOKEN_EXPIRED = "The token has expired."
