--- image/image_utils.py
+++ image/image_utils.py
@@ -32,6 +32,7 @@
 import os
 import re
 import tempfile
+import typing
 from typing import ContextManager, Generator, Optional
 
 import cryptography
@@ -164,20 +165,28 @@
         path: str,
         run_as_root: bool = True,
         force_share: bool = False,
-        allow_qcow2_backing_file: bool = False) -> imageutils.QemuImgInfo:
+        allow_qcow2_backing_file: bool = False,
+        img_format: Optional[str] = None) -> imageutils.QemuImgInfo:
     """Return an object containing the parsed output from qemu-img info."""
 
-    format_name = cinder.privsep.format_inspector.get_format_if_safe(
-        path=path,
-        allow_qcow2_backing_file=allow_qcow2_backing_file)
-    if format_name is None:
-        LOG.warning('Image/Volume %s failed safety check', path)
-        # NOTE(danms): This is the same exception as would be raised
-        # by qemu_img_info() if the disk format was unreadable or
-        # otherwise unsuitable.
-        raise exception.Invalid(
-            reason=_('Image/Volume failed safety check'))
+    format_name = img_format
+    # NOTE(sfernand): In case we are trying to inspect a raw volume containing
+    # a qcow2 image, inspection will incorrectly report the inner (qcow2)
+    # format, instead of the expected outer (raw) format. To avoid that, we
+    # need skip format inspection if the Cinder volume is known to be raw.
+    if img_format != 'raw':
+        format_name = cinder.privsep.format_inspector.get_format_if_safe(
+            path=path,
+            allow_qcow2_backing_file=allow_qcow2_backing_file)
+        if format_name is None:
+            LOG.warning('Image/Volume %s failed safety check', path)
+            # NOTE(danms): This is the same exception as would be raised
+            # by qemu_img_info() if the disk format was unreadable or
+            # otherwise unsuitable.
+            raise exception.Invalid(
+                reason=_('Image/Volume failed safety check'))
 
+    format_name = typing.cast(str, format_name)
     cmd = ['env', 'LC_ALL=C', 'qemu-img', 'info',
            '-f', format_name, '--output=json']
     if force_share:
@@ -222,6 +231,20 @@
     return info
 
 
+def get_image_format(path: str,
+                     allow_qcow2_backing_file: Optional[bool] = False
+                     ) -> Optional[str]:
+    """Inspect image file and return its format if safe.
+
+    :param path: Path to the image file to inspect
+    :param allow_qcow2_backing_file: Allow qcow2 images with backing files
+    :returns: Format name ('raw', 'qcow2', etc.) if safe, None if unsafe
+    """
+    return cinder.privsep.format_inspector.get_format_if_safe(
+        path=path,
+        allow_qcow2_backing_file=allow_qcow2_backing_file)
+
+
 def get_qemu_img_version() -> Optional[list[int]]:
     """The qemu-img version will be cached until the process is restarted."""
 
