--- ovn-24.03.8/northd/northd.c.orig	2026-08-17 12:05:52.814104246 +0800
+++ ovn-24.03.8/northd/northd.c	2026-08-17 12:05:52.841104362 +0800
@@ -15301,7 +15301,7 @@
 
     if (!od->is_gw_router) {
         /* Distributed router. */
-        if (od->n_l3dgw_ports) {
+        if (od->n_l3dgw_ports && distributed_nat) {
             priority += 128;
         }
 
@@ -15346,7 +15346,7 @@
                                  cidr_bits, is_v6, l3dgw_port,
                                  lflow_ref, false);
 
-    if (od->n_l3dgw_ports) {
+    if (od->n_l3dgw_ports && distributed_nat) {
         priority += 128;
     }
 
@@ -15413,7 +15413,7 @@
 
     if (!od->is_gw_router) {
         /* Distributed router. */
-        if (od->n_l3dgw_ports) {
+        if (od->n_l3dgw_ports && distributed_nat) {
             priority += 128;
         }
 
