--- conf/__init__.py
+++ conf/__init__.py
@@ -29,6 +29,7 @@ from nova.conf import conductor
 from nova.conf import configdrive
 from nova.conf import console
 from nova.conf import consoleauth
+from nova.conf import cube
 from nova.conf import cyborg
 from nova.conf import database
 from nova.conf import devices
@@ -76,6 +77,7 @@ conductor.register_opts(CONF)
 configdrive.register_opts(CONF)
 console.register_opts(CONF)
 consoleauth.register_opts(CONF)
+cube.register_opts(CONF)
 cyborg.register_opts(CONF)
 database.register_opts(CONF)
 devices.register_opts(CONF)
