--- openstack/compute/v2/server.py
+++ openstack/compute/v2/server.py
@@ -320,6 +320,11 @@
         if hint_key in server_body:
             request.body[hint_key] = server_body.pop(hint_key)
 
+        # hypervisor_hostname exists with a prefix on response, but not request
+        hostname_key = "OS-EXT-SRV-ATTR:hypervisor_hostname"
+        if hostname_key in server_body:
+            server_body["hypervisor_hostname"] = server_body.pop(hostname_key)
+
         return request
 
     def _action(self, session, body, microversion=None):
